pkg/kerr/kerr.go

```diff
@@ -184,113 +185,125 @@ var (
-	104: InconsistentClusterID,
-	105: TransactionalIDNotFound,
-	106: FetchSessionTopicIDError,
+var code2err = makeCodeMapping()
+
+func makeCodeMapping() map[int16]error {
+	mapping := map[int16]error{0: nil}
+	for _, e := range []*Error{
+		UnknownServerError,
+		OffsetOutOfRange,
+		CorruptMessage,
+		UnknownTopicOrPartition,
+		InvalidFetchSize,
+		LeaderNotAvailable,
+		NotLeaderForPartition,
+		RequestTimedOut,
+		BrokerNotAvailable,
+		ReplicaNotAvailable,
+		MessageTooLarge,
+		StaleControllerEpoch,
+		OffsetMetadataTooLarge,
+		NetworkException,
+		CoordinatorLoadInProgress,
+		CoordinatorNotAvailable,
+		NotCoordinator,
+		InvalidTopicException,
+		RecordListTooLarge,
+		NotEnoughReplicas,
+		NotEnoughReplicasAfterAppend,
+		InvalidRequiredAcks,
+		IllegalGeneration,
+		InconsistentGroupProtocol,
+		InvalidGroupID,
+		UnknownMemberID,
+		InvalidSessionTimeout,
+		RebalanceInProgress,
+		InvalidCommitOffsetSize,
+		TopicAuthorizationFailed,
+		GroupAuthorizationFailed,
+		ClusterAuthorizationFailed,
+		InvalidTimestamp,
+		UnsupportedSaslMechanism,
+		IllegalSaslState,
+		UnsupportedVersion,
+		TopicAlreadyExists,
+		InvalidPartitions,
+		InvalidReplicationFactor,
+		InvalidReplicaAssignment,
+		InvalidConfig,
+		NotController,
+		InvalidRequest,
+		UnsupportedForMessageFormat,
+		PolicyViolation,
+		OutOfOrderSequenceNumber,
+		DuplicateSequenceNumber,
+		InvalidProducerEpoch,
+		InvalidTxnState,
+		InvalidProducerIDMapping,
+		InvalidTransactionTimeout,
+		ConcurrentTransactions,
+		TransactionCoordinatorFenced,
+		TransactionalIDAuthorizationFailed,
+		SecurityDisabled,
+		OperationNotAttempted,
+		KafkaStorageError,
+		LogDirNotFound,
+		SaslAuthenticationFailed,
+		UnknownProducerID,
+		ReassignmentInProgress,
+		DelegationTokenAuthDisabled,
+		DelegationTokenNotFound,
+		DelegationTokenOwnerMismatch,
+		DelegationTokenRequestNotAllowed,
+		DelegationTokenAuthorizationFailed,
+		DelegationTokenExpired,
+		InvalidPrincipalType,
+		NonEmptyGroup,
+		GroupIDNotFound,
+		FetchSessionIDNotFound,
+		InvalidFetchSessionEpoch,
+		ListenerNotFound,
+		TopicDeletionDisabled,
+		FencedLeaderEpoch,
+		UnknownLeaderEpoch,
+		UnsupportedCompressionType,
+		StaleBrokerEpoch,
+		OffsetNotAvailable,
+		MemberIDRequired,
+		PreferredLeaderNotAvailable,
+		GroupMaxSizeReached,
+		FencedInstanceID,
+		EligibleLeadersNotAvailable,
+		ElectionNotNeeded,
+		NoReassignmentInProgress,
+		GroupSubscribedToTopic,
+		InvalidRecord,
+		UnstableOffsetCommit,
+		ThrottlingQuotaExceeded,
+		ProducerFenced,
+		ResourceNotFound,
+		DuplicateResource,
+		UnacceptableCredential,
+		InconsistentVoterSet,
+		InvalidUpdateVersion,
+		FeatureUpdateFailed,
+		PrincipalDeserializationFailure,
+		SnapshotNotFound,
+		PositionOutOfRange,
+		UnknownTopicID,
+		DuplicateBrokerRegistration,
+		BrokerIDNotRegistered,
+		InconsistentTopicID,
+		InconsistentClusterID,
+		TransactionalIDNotFound,
+		FetchSessionTopicIDError,
+		IneligibleReplica,
+		NewLeaderElected,
```

Comment:
I'm actually not sure I'm too positive on this change -- the prior code makes it easy to see that I'm missing errors (currently 107 and 108, actually).

If we go with this change, I'd like this error list to be ordered to more easily visually see that errors are missing. that said, I do think the older list is easier to maintain -- and it changes so rarely that the extra bit of duplication in the numbers isn't really a maintenance burden.